node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeAllocation.java

```diff
@@ -289,15 +286,28 @@ private Node acceptNode(NodeCandidate candidate, Retirement retirement, boolean
             node = node.retire(nodeRepository.clock().instant());
         }
         if ( ! node.allocation().get().membership().cluster().equals(cluster)) {
-            // group may be different
-            node = setCluster(cluster, node);
+            // Cluster has the updated settings but do not set a group
+            node = setCluster(cluster.with(node.allocation().get().membership().cluster().group()), node);
         }
-        candidate = candidate.withNode(node);
+        candidate = candidate.withNode(node, retirement != Retirement.none && retirement != Retirement.alreadyRetired );
         indexes.add(node.allocation().get().membership().index());
         nodes.put(node.hostname(), candidate);
         return node;
     }
 
+    private boolean canBeUsedInGroupWithDeficiency(Node node) {
+        if (requestedNodes.count().isEmpty()) return true;
+        if (node.allocation().isEmpty()) return true;
+        var group = node.allocation().get().membership().cluster().group();
+        if (group.isEmpty()) return true;
+        long nodesInGroup = nodes.values().stream().filter(n -> groupOf(n).equals(group)).count();
+        return nodesInGroup < requestedNodes.count().get() / requestedNodes.groups();
```

Comment:
Not sure I understand this... Seems like we want to return true if the group the given `node` is in already has more nodes than what is requested, then this node can instead by reallocated to a different group? If so, then shouldn't the sign be reversed?

Reply:
The premise is: We cannot _change_ the group of a node we will keep (unless that group is to be removed), because that will lead to temporary coverage loss.

This method returns whether - given that - this node can be used in a group which has to few nodes. If the group the node belongs to has fewer nodes than it should, it can.